tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -497,32 +630,67 @@ class SPIRVEmitter : public ASTConsumer {
     } else if (auto *castExpr = dyn_cast<ImplicitCastExpr>(expr)) {
       const uint32_t fromValue = doExpr(castExpr->getSubExpr());
       // Using lvalue as rvalue will result in a ImplicitCast in Clang AST.
-      // This place gives us a place to inject the code for handling interface
-      // variables. Since using the <result-id> of an interface variable as
+      // This place gives us a place to inject the code for handling stage
+      // variables. Since using the <result-id> of a stage variable as
       // rvalue means OpLoad it first. For normal values, it is not required.
-      if (declIdMapper.isInterfaceVariable(fromValue)) {
+      if (declIdMapper.isStageVariable(fromValue)) {
         const uint32_t resultType =
-            translateType(castExpr->getType(), theBuilder);
+            typeTranslator.translateType(castExpr->getType());
         return theBuilder.createLoad(resultType, fromValue);
       }
       return fromValue;
+    } else if (auto *memberExpr = dyn_cast<MemberExpr>(expr)) {
+      const uint32_t base = doExpr(memberExpr->getBase());
+      auto *memberDecl = memberExpr->getMemberDecl();
+      if (auto *fieldDecl = dyn_cast<FieldDecl>(memberDecl)) {
+        const auto index = theBuilder.getInt32Value(fieldDecl->getFieldIndex());
+        const uint32_t fieldType =
+            typeTranslator.translateType(fieldDecl->getType());
+        const uint32_t ptrType =
+            theBuilder.getPointerType(fieldType, spv::StorageClass::Function);
+        return theBuilder.createAccessChain(ptrType, base, {index});
```

Comment:
I believe so since the above is just `MemberExpr` -> `MemberExpr` -> `DeclRefExpr`. But testing it is out of the scope of this PR. We'll definitely need to check this case when doing nested structs.